### PR TITLE
Prevent duplicate declarations of packages (II)

### PR DESCRIPTION
**Pull request by wouteroostervld**
Fix test

### CODE REVIEW

**Comment by wouteroostervld** [comment]
Tests failed in #83.

File: manifests/packages.pp [hunk 1]
```diff
@@ -1,12 +1,15 @@
 class postfix::packages {
   include ::postfix::params
 
-  package { 'postfix':
-    ensure => installed,
+  if ! defined(Package['postfix']) {
```
Comment (mcanevet) [inline manifests/packages.pp]: Managing the postfix package resource is totally in the scope of this module, so I can't see the point of checking for its declaration first...

File: manifests/packages.pp [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+    package { 'postfix':
+      ensure => installed,
+    }
   }
 
-  package { 'mailx':
-    ensure => installed,
-    name   => $postfix::params::mailx_package,
+  if ! defined(Package[$postfix::params::mailx_package]) {
```
Comment (mcanevet) [inline manifests/packages.pp]: Same here

**Comment by raphink** [comment]
I will close this PR for now. Please consider @mcanevet's comments. One acceptable way to achieve what you want would be to add `$package_name` and `$package_ensure` parameters to the class to override the defaults.

**Pull request closed without merging**